digits/scheduler.py

```diff
@@ -246,37 +242,37 @@ def delete_job(self, job):
             raise ValueError('called delete_job with a %s' % type(job))
         dependent_jobs = []
         # try to find the job
-        for i, job in enumerate(self.jobs):
-            if job.id() == job_id:
-                if isinstance(job, DatasetJob):
-                    # check for dependencies
-                    for j in self.jobs:
-                        if isinstance(j, ModelJob) and j.dataset_id == job.id():
-                            logger.error('Cannot delete "%s" (%s) because "%s" (%s) depends on it.' % (job.name(), job.id(), j.name(), j.id()))
-                            dependent_jobs.append(j.name())
-                if len(dependent_jobs)>0:
-                    error_message = 'Cannot delete "%s" because %d model%s depend%s on it: %s' % (
-                            job.name(),
-                            len(dependent_jobs),
-                            ('s' if len(dependent_jobs) != 1 else ''),
-                            ('s' if len(dependent_jobs) == 1 else ''),
-                            ', '.join(['"%s"' % j for j in dependent_jobs]))
-                    raise errors.DeleteError(error_message)
-                self.jobs.pop(i)
-                job.abort()
-                if os.path.exists(job.dir()):
-                    shutil.rmtree(job.dir())
-                logger.info('Job deleted.', job_id=job_id)
-                from digits.webapp import socketio
-                socketio.emit('job update',
-                              {
-                                  'update': 'deleted',
-                                  'job_id': job.id()
-                              },
-                              namespace='/jobs',
-                              room='job_management',
-                )
-                return True
+        job = self.get_job(job_id)
+        if job:
+            if isinstance(job, DatasetJob):
+                # check for dependencies
+                for j in self.jobs:
+                    if isinstance(j, ModelJob) and j.dataset_id == job.id():
+                        logger.error('Cannot delete "%s" (%s) because "%s" (%s) depends on it.' % (job.name(), job.id(), j.name(), j.id()))
+                        dependent_jobs.append(j.name())
+            if len(dependent_jobs)>0:
+                error_message = 'Cannot delete "%s" because %d model%s depend%s on it: %s' % (
+                        job.name(),
+                        len(dependent_jobs),
+                        ('s' if len(dependent_jobs) != 1 else ''),
+                        ('s' if len(dependent_jobs) == 1 else ''),
+                        ', '.join(['"%s"' % j for j in dependent_jobs]))
+                raise errors.DeleteError(error_message)
+            del self.jobs[job_id]
```

Comment:
pop would avoid a potential KeyError, which in this case should not happen, because of the `job = self.jobs.get(job_id, None)`.  However it's a better practice.  I'll make that change.